File: 0206-reverse-linked-list/0206-reverse-linked-list.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val = 0, _next = nil)
#         @val = val
#         @next = _next
#     end
# end
# @param {ListNode} head
# @return {ListNode}
def reverse_list(head)
   @new_head=[]
    def get_tail(node)
        return [] if node.nil?
        if node.next.nil?
            @new_head=ListNode.new(node.val)
            return @new_head
        end
        new_node=ListNode.new(node.val)
        get_tail(node.next).next=new_node
        new_node
    end
    get_tail(head)
    @new_head
end

File: 0206-reverse-linked-list/0206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {

    if(!head||!head.next)
    return head

        let head1=head
        let head2=head.next;
        head1.next=null;
        do {
            temp=head2.next
            head2.next=head1;
            head1=head2;
            head2=temp;
            // console.log(head1, head2)
        }while(head2)

        return head1;
};